client/login.go

```diff
@@ -79,29 +80,53 @@ func (client *Client) LoggedIn() bool {
 }
 
 func storeAuthDataFilename() string {
-	authFilename := os.Getenv("SNAPPY_STORE_AUTH_DATA_FILENAME")
-	if authFilename != "" {
-		return authFilename
-	}
-	homeDir, err := osutil.CurrentHomeDir()
+	real, err := osutil.RealUser()
 	if err != nil {
 		panic(err)
 	}
+
+	return storeAuthDataFilenameFromHome(real.HomeDir)
+}
+
+const authFileEnvKey = "SNAPPY_STORE_AUTH_DATA_FILENAME"
+
+func storeAuthDataFilenameFromHome(homeDir string) string {
+	if fn := os.Getenv(authFileEnvKey); fn != "" {
```

Comment:
This looks like a bug given the function notation. We explicitly ask for it to be in a given place, yet it goes somewhere else entirely. This whole new logic also seems to be fixing something that wasn't broken. The home directory was being picked correctly before already. Can we go back to what it was before?